### PR TITLE
Add renderer + extract parser into its own gem.

### PR DESCRIPTION
**Pull request by beauby**
+ Add a low-level renderer.
+ Break the gem into two subgems (jsonapi-parser and jsonapi-renderer).

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
:+1:

File: jsonapi.gemspec [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version = File.expand_path('../JSONAPI_VERSION', __FILE__).strip
```
Comment (NullVoxPopuli) [inline jsonapi.gemspec]: I've normally seen this as a constant in `lib/gem_name/version.rb` what's the reason for doing it this way?

File: lib/jsonapi.rb [hunk 2]
```diff
@@ -1,18 +1,2 @@
...
-require 'jsonapi/relationships'
-require 'jsonapi/resource'
-require 'jsonapi/resource_identifier'
-
-require 'jsonapi/parse'
-
-module JSONAPI
-end
+require 'jsonapi/parser'
+require 'jsonapi/renderer'
```
Comment (NullVoxPopuli) [inline lib/jsonapi.rb]: :+1:

File: jsonapi.gemspec (hunk 1 above)
Comment (beauby) [inline jsonapi.gemspec]: Well here, the base `jsonapi` gem would have to be re-released each time either of the subgems gets a new release, so as a quick hack, I thought I'd keep the 3 in sync at the beginning. Think of it like this: there is fundamentally one gem, but as a convenience you can depend on only a part of it. That's the versioning scheme Rails uses for its various components (activesupport, actionpack, etc.).

Comment (NullVoxPopuli) [inline jsonapi.gemspec]: yeah, I was just looking at the way Rails did it: https://github.com/rails/rails/blob/master/version.rb

neat . :-)

Comment (beauby) [inline jsonapi.gemspec]: Yeah but the `version.rb` is just a programmatic API for accessing the version through runtime (which currently does not seem to add much value to this gem). It's always possible to add it later.

**Comment by NullVoxPopuli** [comment]
yay passing tests!!!! 🎉 🎈

**Pull request merged**